elasticapm/instrumentation/packages/botocore.py

```diff
@@ -214,6 +214,7 @@ def modify_span_sqs(span, args, kwargs):
         else:
             messages = []
         for message in messages:
+            message["MessageAttributes"] = message.get("MessageAttributes") or {}
```

Comment:
Is `message` not a normal dict? Or is the key defined as `None`, maybe? Just thought it looked a little odd, but I certainly might be missing something.

```suggestion
            message["MessageAttributes"] = message.get("MessageAttributes", {})
```

Reply:
It's just a silly little optimization. The thinking was that with `.get("foo", {})`, a new `dict` is always instantiated and subsequently thrown away if the key is found. With `or`, the dict is only instantiated if the lookup fails, thanks to short-circuiting. A quick benchmark using `timeit`:

    In [1]: message = {"a": 1}
    In [2]: never_get = lambda: message.get("a", {})
    In [3]: never_or = lambda: message.get("a") or {}
    In [4]: always_get = lambda: message.get("b", {})
    In [5]: always_or = lambda: message.get("b") or {}
    In [6]: import timeit
    In [7]: timeit.timeit(never_get, setup="gc.enable()")
    Out[7]: 0.09470749995671213
    In [8]: timeit.timeit(never_or, setup="gc.enable()")
    Out[8]: 0.08076889999210835
    In [9]: timeit.timeit(always_get, setup="gc.enable()")
    Out[9]: 0.09426169993821532
    In [10]: timeit.timeit(always_or, setup="gc.enable()")
    Out[10]: 0.09646200004499406

So, using `or` is somewhat faster when the key is there most of the time. It's only a difference of ~0.01us though, so if you prefer the slightly less confusing code, I'm ok with that.

Reply:
Interesting! Thanks for the explanation!